src/main/java/com/metamx/common/io/smoosh/FileSmoosher.java

```diff
@@ -160,7 +174,7 @@ public SmooshedWriter addWithSmooshedWriter(final String name, final long size)
     }
 
     final int startOffset = currOut.getCurrOffset();
-
+    useDelegateSmooshWriter = true;
```

Comment:
Either the name of the outright usage of a boolean like this makes it really hard to understand why you are using the delegate.  What's the specific set of circumstances under which you want to use it?  Either name the boolean after those circumstances or just use that logic when determining what you want to do.